src/Utils.jl

```diff
@@ -172,10 +172,12 @@ function ssa_path_cost(ssa_path, inputs, output, size)
         a, b = inputs[i], inputs[j]
         inds_ij = symdiff(a, b) ∪ ∩(output, a, b)
         flops_ij = flops(a, b, size, output)
-        append!(inputs, inds_ij)
+        append!(inputs, [inds_ij])
 
         cost += flops_ij
-        max_size = max(max_size, mapreduce(ind -> size[ind], *, inds_ij))
+        if (length(inds_ij) > 0)
+            max_size = max(max_size, mapreduce(ind -> size[ind], *, inds_ij))
+        end
```

Comment:
Why do you need this to be conditional?

Reply:
It can be fixed by adding the `init` keyword argument like this:
```julia
max_size = max(max_size, mapreduce(ind -> size[ind], *, inds_ij; init=1))
```

That way, the accumulator variable of the reduction is defined even if the iterator (i.e. `ind_ij`) is empty.

Reply:
Fantastic, thanks. I added this to the last commit, together with the fix you suggested for `max_size`.